multicluster/controllers/multicluster/member_clusterset_controller.go

```diff
@@ -207,8 +212,14 @@ func (r *MemberClusterSetReconciler) createOrUpdateRemoteCommonArea(clusterSet *
 		return err
 	}
 
-	r.remoteCommonArea, err = commonarea.NewRemoteCommonArea(clusterID, r.clusterSetID, common.ClusterSetID(r.clusterID), url, secret, r.Scheme,
-		r.Client, clusterSet.Spec.Namespace, r.Namespace)
+	config, remoteCommonAreaMgr, remoteClient, err := r.GetRemoteCommonAreaPre(secret, url, clusterID, clusterSet)
```

Comment:
Why do you need to move `GetRemoteCommonAreaPre` logic to member_clusterset_controller? You can still keep those codes in the common area and call the commonarea's function.
You may defined a func variable like `getRemoteConfigAndClient = commonarea.GetRemoteConfigAndClient` at`member_clusterset_controller`, and move `GetRemoteCommonAreaPre` back to common area package and rename it as `GetRemoteConfigAndClient`. Then you can assign `getRemoteConfigAndClient` with fake function in the test file to use fake client.